Automattic/wp-calypso
# Editor: use EditorTermSelector for categories

Part of #3697: This branch seeks to use `<EditorTermSelector />` for post categories in the editor sidebar.  Currently, categories are managed via the legacy flux-powered `<CategorySelector />`.

This branch also `connect()`s the `<EditorCategoriesTags />` component to the state tree, moves category persistence to the `terms` object - and removes legacy logic from the `post-edit-store` that normalizes `category_ids`.

__Testing__
- Open a new post in the editor
- Expand the tags & categories accordion in the sidebar
- Interact with the category selector, search, select/unselect items
- Make sure the label on the accordion updates as expected when you select a category(ies)
- Save the post and/or publish and verify your selections are persisted
- Add a new category ( if your user has `manage_categories` capability ).  Verify the new category is added and auto-selected

__Todo / Questions__
- [x] Add logic to toggle the display of "Add Term" based upon capabilities.  There is a check in place right now in `<EditorCategories />` but we need to have similar logic for all taxonomy types that use `<EditorTermSelector />`
- [x] The accordion tests are now broken since the component has been tied to the redux state tree.  Should the label creation logic be moved to a selector or a util to ease the testability of it?  Thoughts @nylen @aduth 

__Capability Issue__
In the terms API endpoint we check for the `edit_terms` capability.  Currently, this capability is not exposed on the user object so I have continued using `manage_categories` to conditionally show the Add Term button.

__UX Question__
The current approach taken here also fixes #561 - where the "Default Category" is never selected in the <EditorTermSelector /> on a new draft.  The label shows the default category name, but until an autosave, or manual save happens, no checkbox is toggled:

<img width="257" alt="new_post_ _testingtimmy2_wordpress_com_ _wordpress_com" src="https://cloud.githubusercontent.com/assets/22080/17368318/4928b040-5948-11e6-8ce2-9d31f6c29d26.png">

Additionally, if all categories are un-checked, the default category will not be checked again until the next Save/Update payload returns from the API.  This mimics the flow in core but /cc @mtias to see if this is the behavior you were looking for.

Test live: https://calypso.live/?branch=update/editor/category-to-term-selector
